src/graph.rs

```diff
@@ -1287,7 +1281,7 @@ impl<T> Graph<T>
     /// 
     /// This function will return a default label if no label is set. Returns
     /// `None` if there is no vertex associated with the given id in the graph.
-    pub fn label(&self, vertex_id: &VertexId) -> Option<String> {
+    pub fn label(&mut self, vertex_id: &VertexId) -> Option<String> {
```

Comment:
The reason `mem::transmute` is being used is because from an api perspective this method is a read operation with interior mutability i.e. `&self` instead of `&mut self`. 

This is intended since for example if you call `Graph::label` on a graph behind a `RwLock` you would have to lock it in write position.

While I agree that `mem::transmute` should not be used I do not agree with changing the api from receiving a `&mut self` instead of `&self`.

If you wish to remove `mem::transmute` we need to find a way to achieve interior mutability without:
a) Using `RefCell` which makes the `Graph` not `Send`.
b) Using locks which induce overhead.

Reply:
I absolutely agree that public api does not expected to get it with `&mut self`.About the changes you insisted on, that additional allocation should be since there will be a problem with `link` and `mut link` on `self`.

Sorry for bare contribution, and your time :)